app/models/barbeque/job_definition.rb

```diff
@@ -15,12 +15,25 @@ class Barbeque::JobDefinition < Barbeque::ApplicationRecord
   DATE_HOUR_SQL = 'date_format(created_at, "%Y-%m-%d %H:00:00")'
 
   def execution_stats(from, to)
-    job_executions.where(created_at: from .. to).group(DATE_HOUR_SQL).order(DATE_HOUR_SQL).pluck("#{DATE_HOUR_SQL}, count(1), avg(timestampdiff(second, created_at, finished_at))").map do |date_hour, count, avg_time|
-      {
-        date_hour: Time.zone.parse("#{date_hour} UTC"),
+    from = from.beginning_of_hour
+    to = to.beginning_of_hour
+    stats = {}
+    job_executions.where(created_at: from .. to).group(DATE_HOUR_SQL).order(DATE_HOUR_SQL).pluck("#{DATE_HOUR_SQL}, count(1), avg(timestampdiff(second, created_at, finished_at))").each do |date_hour, count, avg_time|
+      time = Time.zone.parse("#{date_hour} UTC")
+      stats[time] = {
         count: count,
         avg_time: avg_time,
       }
     end
+    t = from
+    while t < to
+      stats[t] ||= {
+        count: 0,
+        avg_time: 0,
+      }
+      t += 1.hour
+    end
+
+    stats.keys.sort.map { |time| stats[time].merge(date_hour: time) }
```

Comment:
I have no preference.
I've slightly changed the suggested code:

- Use block to set the default value
- Use `...` to exclude `to` value

https://github.com/cookpad/barbeque/pull/12/commits/c63c2c4bb5663a692145b08e29d6487e23d78aa5